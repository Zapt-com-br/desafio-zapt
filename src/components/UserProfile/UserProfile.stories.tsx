import React from 'react';

import { Story, Meta } from '@storybook/react/types-6-0';
import UserProfile, { UserProfileProps } from '.';

export default {
  title: 'Social Comments/UserProfile',
  component: UserProfile,
} as Meta;

export const UserProfileDefault: Story<UserProfileProps> = args => (
  <UserProfile {...args} />
);
UserProfileDefault.args = {
  username: '@LuisCarlos',
  subtext: 'São Paulo • São Paulo',
  imageProfile:
    'https://www.pixsy.com/wp-content/uploads/2021/04/ben-sweet-2LowviVHZ-E-unsplash-1.jpeg',
};
