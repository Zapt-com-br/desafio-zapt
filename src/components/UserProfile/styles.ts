import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
`;

export const TextWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Image = styled.img`
  border-radius: 50%;
  width: 3.6rem;
  height: 3.6rem;
  object-fit: cover;
  margin-right: 1.2rem;
`;
