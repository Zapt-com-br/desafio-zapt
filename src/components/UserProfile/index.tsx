import React from 'react';

import { Typography } from 'components';

import * as Styled from './styles';

type Sizes = 'small' | 'medium' | 'xmedium' | 'large';
type Colors = 'black' | 'darkgray' | 'white' | 'primary';

export type UserProfileProps = {
  username: string;
  subtext?: string;
  imageProfile: string;
  textColor?: Colors;
  usernameSize?: Sizes;
};

const UserProfile = ({
  imageProfile,
  username,
  textColor,
  subtext = '',
  usernameSize = 'medium',
}: UserProfileProps) => (
  <Styled.Wrapper>
    <Styled.Image src={imageProfile} alt="Image profile" />
    <Styled.TextWrapper>
      <Typography bold="500" size={usernameSize} color={textColor}>
        {username}
      </Typography>
      <Typography size="xsmall" color={textColor}>
        {!!subtext && subtext}
      </Typography>
    </Styled.TextWrapper>
  </Styled.Wrapper>
);

export default UserProfile;
