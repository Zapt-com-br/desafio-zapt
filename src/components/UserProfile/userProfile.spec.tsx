import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithTheme } from 'helpers/renderWithThame';

import UserProfile, { UserProfileProps } from '.';

const props: UserProfileProps = {
  username: '@LuisCarlos',
  imageProfile:
    'https://www.pixsy.com/wp-content/uploads/2021/04/ben-sweet-2LowviVHZ-E-unsplash-1.jpeg',
};

describe('<UserProfile />', () => {
  it('should render UserProfile', () => {
    const { container } = renderWithTheme(<UserProfile {...props} />);

    const imgProfile = screen.getByRole('img', { name: /image profile/i });
    const username = screen.getByText(props.username);

    expect(imgProfile).toBeInTheDocument();
    expect(username).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render UserProfile whith subtext', () => {
    renderWithTheme(<UserProfile subtext="subtext" {...props} />);

    const element = screen.getByText(/subtext/i);

    expect(element).toBeInTheDocument();
  });

  it('should render UserProfile whithout subtext', () => {
    renderWithTheme(<UserProfile {...props} />);

    const element = screen.queryByText(/subtext/i);

    expect(element).not.toBeInTheDocument();
  });
});
