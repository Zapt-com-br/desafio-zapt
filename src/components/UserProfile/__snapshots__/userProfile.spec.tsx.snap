// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> should render UserProfile 1`] = `
.c3 {
  font-weight: 500;
  color: #000000;
  font-size: 1.6rem;
}

.c4 {
  font-weight: 400;
  color: #000000;
  font-size: 1.2rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border-radius: 50%;
  width: 3.6rem;
  height: 3.6rem;
  object-fit: cover;
  margin-right: 1.2rem;
}

<div
  class="c0"
>
  <img
    alt="Image profile"
    class="c1"
    src="https://www.pixsy.com/wp-content/uploads/2021/04/ben-sweet-2LowviVHZ-E-unsplash-1.jpeg"
  />
  <div
    class="c2"
  >
    <span
      class="c3"
      color="black"
    >
      @LuisCarlos
    </span>
    <span
      class="c4"
      color="black"
    />
  </div>
</div>
`;
